python/src/aiconfig/editor/server/server.py

```diff
@@ -162,24 +171,113 @@ def create() -> FlaskResponse:
+        def run_async_config_in_thread():
+            asyncio.run(aiconfig.run(
+                    prompt_name=prompt_name,
+                    params=params,
+                    run_with_dependencies=False,
+                    options=inference_options,
+                )
+            )
+            output_text_queue.put(STOP_STREAMING_SIGNAL)
+        t = threading.Thread(target=run_async_config_in_thread)
+        t.start()
+
+        # Create a deep copy of the state aiconfig so we can yield an AIConfig
+        # with streaming partial outputs in the meantime. This probably isn't
+        # necessary, but just getting unblocked for now
+        displaying_config = copy.deepcopy(aiconfig)
+
+        # Need to wait until streamer has at least 1 item to display
+        SLEEP_DELAY_SECONDS = 0.1
+        MAX_TIMEOUT_SECONDS = 5.0
+        wait_time_in_seconds = 0.0
+        # TODO: add an "and" for t.isAlive() so that we break if t is complete on non-streaming
+        while output_text_queue.isEmpty():
+            time.sleep(0.1)
+            wait_time_in_seconds += SLEEP_DELAY_SECONDS
+            print(f"Output queue is currently empty. Waiting for {wait_time_in_seconds:.1f}s...")
+
+            # TODO: We should have a better way to check if the model supports
+            # streaming or not and bypass this if they do. I'm thinking we
+            # could add an abstract field that all models need to set T/F
+            # but we'll see. For now this works
+            # And yea I know time.sleep() isn't super accurate, but it's fine,
+            # we can fix later
+            if wait_time_in_seconds >= MAX_TIMEOUT_SECONDS:
+                print(f"Output queue is still empty after {wait_time_in_seconds:.1f}s. Breaking...")
+                break
+
+        yield "["
+        aiconfig_json: str | None = None
+        if not output_text_queue.isEmpty():
+            accumulated_output_text = ""
+            for text in output_text_queue:
+                if isinstance(text, str):
+                    accumulated_output_text += text
+                elif isinstance(text, dict) and "content" in text:
+                    # TODO: Fix streaming output format so that it returns text
+                    accumulated_output_text += text["content"]
+
+                accumulated_output : Output = ExecuteResult(
+                    **{
+                        "output_type": "execute_result",
+                        "data": accumulated_output_text,
+                        # Assume streaming only supports single output
+                        # I think this actually may be wrong for PaLM or OpenAI
+                        # TODO: Need to sync with Ankush but can fix forward
+                        "execution_count": 0, 
+                        "metadata": {},
+                    }
+                )
+
+                displaying_config.add_output(prompt_name, accumulated_output, overwrite=True)
```

Comment:
I realized that this will fail for run_with_dependencies since we don't have a way of tracking the currently executing prompt, and just overwriting the last prompt name instead. Since we are deep copying the aiconfig vs. `displaying_config`, the final result won't be as bad, but it'll look weird during streaming and we should fix. I have a pseudo-diff in https://github.com/lastmile-ai/gradio-workbook/pull/64 with two separate proposals (backend vs. frontend) and would like feedback on it from others